04/app.js

```diff
@@ -5,4 +5,69 @@ const menuItems = [
     {text: 'start', url: '/'},
     {text: 'galeria', url: '/gallery'},
     {text: 'kontakt', url: '/contact'},
-];
\ No newline at end of file
+];
+
+
+// Sposób drugi
+
+const nav = document.querySelector('nav');
+console.log(nav);
+
+if(nav) {
+    const ul = document.createElement('ul');
+
+    if(ul) {
+        for(let i=1; i<=3; i++) {
+            ul.appendChild(  document.createElement('li') );
+        }
+    }
+
+    nav.appendChild(ul)
+    nav.childNodes[1].remove();
+
+    document.querySelectorAll('li').forEach( function(element, index) {
+
+        element.textContent = menuItems[index].text;
+        element.setAttribute( 'url', menuItems[index].url );
+    } )
```

Comment:
Zobacz, że masz tutaj ponowne wyszukanie elementów.
Mógłbyś ustawić treść od razu w pętli `for`.
PS. Element `<li>` powinien mieć jeszcze `<a>` i to ten element będzie linkiem.

Reply:
Faktycznie, nie doczytałem, że w treści zadania jest podane "a". Nawet zapisałem sobie, żeby zapytać Ciebie dlaczego li miało być linkiem :D. Co do zadanka to poprawiłem, mam nadzieję że o to chodziło.